### DISCUSSION TITLE
primaryDisplay on relationship value

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#3895] [Help] [answered]
Hello,

I have two questions on the handlebars/javascript to display relation values.
- In this example, I want to display the primaryDisplay for the user in a screen:
Locality: {{ habitant_view.habitants.locality }}
This output:
Locality: [{"primaryDisplay":"Schaerbeek - Schaarbeek","_id":"%5B21%5D"}]
Which is not great, so I'm using js to display the name:
return "Locality: " + $("habitant_view.habitants.locality")[0].primaryDisplay;
This output:
Locality: Schaerbeek - Schaarbeek

Is this the intended behaviour? Is there a better way to do this?

- The click to update the text in a screen is great but when an element is set with handlebars or javascript, the content is replaced with the example value and the original content is lost. Maybe when editing such element, it could open the current field content (js or handlebars accordingly), or at least display an alert or something? In case my colleague want to update some texts or correct a typo, they could use the inline editing without loosing the formula/js.

Thank you!

**Comment by xnohat** [comment, marked as answer]
I have same problem, dont know how to access relationship value properly ?

Update:
in handlebars we can access relationship value using pluck helper

{{pluck New Repeater.Workflow Steps.Step Approver 'primaryDisplay'}}

New Repeater.Workflow Steps.Step Approver is relationship column (it's also work in case column name have whitespace)

**Reply by mjashanks** [reply]
In handlebars, its `{{ Repeater.Deals.Contacts.0.primaryDisplay }}` - but pluck will work also.
